lib/index.js

```diff
@@ -46,13 +46,18 @@ exports.buildConnections = (routeStops, routeState) => {
     }, []);
 };
 exports.calculateMetricsSums = (currentConnections, deletedConnections, addedConnections) => {
-    return currentConnections
+    const resultingConnections = currentConnections
         .filter((connection) => !deletedConnections.find((deleted) => deleted === connection.id))
-        .concat(addedConnections)
-        .reduce((sumDurationAndDistance, connection) => ({
-        duration: sumDurationAndDistance.duration + connection.duration,
-        distance: sumDurationAndDistance.distance + connection.distance,
-        stopsCount: sumDurationAndDistance.stopsCount + 1
-    }), { duration: 0, distance: 0, stopsCount: 1 });
+        .concat(addedConnections);
+    if (resultingConnections.length === 0) {
+        return { duration: 0, distance: 0, stopsCount: 0 };
```

Comment:
But how do you know here, that there is no stop? Isn't that the problem that you can have no connections but one stop? How is this case handled now?